src/Symfony/Component/HttpFoundation/Request.php

```
@@ -252,12 +252,24 @@ public function initialize(array $query = [], array $request = [], array $attrib
     public static function createFromGlobals(): static
     {
         $request = self::createRequestFromFactory($_GET, $_POST, [], $_COOKIE, $_FILES, $_SERVER);
-
-        if (str_starts_with($request->headers->get('CONTENT_TYPE', ''), 'application/x-www-form-urlencoded')
-            && \in_array(strtoupper($request->server->get('REQUEST_METHOD', 'GET')), ['PUT', 'DELETE', 'PATCH'], true)
+        $contentType = $request->headers->get('CONTENT_TYPE', '');
+        $method = strtoupper($request->server->get('REQUEST_METHOD', 'GET'));
+
+        if (
+            \in_array($method, ['PUT', 'DELETE', 'PATCH'], true)
+            && (
+                str_starts_with($contentType, 'application/x-www-form-urlencoded')
```

I'd let this check on the content-type to the request_parse_body function, no need to implement it ourselves

---

What do you mean?
We need to check whether it is a non-POST request. For a POST request, the result of `request_parse_body()` seems to be empty. A content type check is also required because if, for example, a request with content-type `application/json` is passed to the function, it will throw `RequestParseBodyException: Content-Type "application/json" is not supported`.

---

I mean the check for the content-type: we can catch the exception and deal with it instead of checking the content-type ourselves (and diverging from the native check).
We have to add logic to deal with the exception anyway, we cannot let it bubble down unhandled (createFromGlobals isn't expected to throw such exceptions).

---

Sorry, my bad. I overlooked the "on the content-type" in your message...
I checked this because `RequestParseBodyException` is also thrown with invalid content, not just unsupported Content-Type. Or are going to differentiate this?